packages/waku/src/lib/plugins/vite-plugin-rsc-analyze.ts

```diff
@@ -34,17 +34,38 @@ export function rscAnalyzePlugin(
           syntax: 'typescript',
           tsx: ext.endsWith('x'),
         });
+        // to fix the issue that the directive is not at the top of the file
+        let isDirective = true;
+        
         for (const item of mod.body) {
           if (
             item.type === 'ExpressionStatement' &&
             item.expression.type === 'StringLiteral'
           ) {
             if (item.expression.value === 'use client') {
-              clientEntryCallback(id);
-              fileHashMap.set(id, await hash(code));
+              if (!isDirective) {
+                const e = {
+                  messageText:
+                    'The `"use client"` directive must be put at the top of the file.',
```

Comment:
Is there any requirement for this?

What's the exact problem you faced?

Reply:
I haven’t faced any issues related to this. I’m just doing this to make sure the compiled code follows React’s guidelines for the use-client directive: https://react.dev/reference/react/use-client#use-client.

In the current code, writing it like this also work like this:
```typescript
import { useState } from 'react';


export const Counter = () => {
  const [count, setCount] = useState(0);
  'use server';

  const handleIncrement = () => setCount((c) => c + 1);

  return (
    <section className="border-blue-400 -mx-4 mt-4 rounded border border-dashed p-4">
      <div>Count: {count}</div>
      <button
        onClick={handleIncrement}
        className="rounded-sm bg-black px-2 py-0.5 text-sm text-white"
      >
        Increment
      </button>
    </section>
  );
};

```

Reply:
Yeah, because inline directive aren't supported.
There's an issue that Vite adds import statements before the directive.

> 'use client' must be at the very beginning of a file, above any imports or other code (comments are OK). They must be written with single or double quotes, but not backticks.

Anyway, I will work on it later. Thanks for sharing your thought.

Reply:
Sorry, there’s a bit of a problem with my description. The way I’ve written it might cause the entire component to be mistakenly recognized as a server component and packaged. I hope this issue can be fixed in future versions, thx

Reply:
> might cause the entire component to be mistakenly recognized as a server component

Currently, no, because we only check top-level directives.
But, in the future when we support inline directives, it might be a server action.

Anyway, if we only use beginning directive as you suggest, it should be fine. (I'm not sure if we error it, maybe eslint rule should warn it.)